From b2d7c0ba042b9eb42fc6fe2de4f471cb86910da8 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Mon, 26 Jul 2021 13:18:42 +0300
Subject: [PATCH] [SW-45031] gaudi2: avoid asic specific duplication

In this patch, Gaudi2 asic HL_TARGET specific
common code moved to common dir.

Change-Id: Id1e1371e49000dce16eb462c015b7b98c451b335
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/app/recovery/Makefile b/app/recovery/Makefile
index 0700cd7..e700ee2 100644
--- a/app/recovery/Makefile
+++ b/app/recovery/Makefile
@@ -27,6 +27,7 @@
 RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/../common/common.c)
 RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/common.c)
 RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/$(HL_TARGET).c)
+RECOVERY_SRCS_C += $(shell ls ../../$(HL_DEV_ARCH)/../common/$(HL_TARGET).c)
 RECOVERY_APP_S := $(shell ls crt0_recovery.S)
 
 RECOVERY_OBJECTS := $(RECOVERY_SRCS_C:.c=.o)
diff --git a/arch/gaudi2/arc/asic.c b/arch/gaudi2/arc/asic.c
index 928a32e..6ec54d8 100644
--- a/arch/gaudi2/arc/asic.c
+++ b/arch/gaudi2/arc/asic.c
@@ -7,12 +7,10 @@
 
 #include <gaudi2_blocks_linux_driver.h>
 #include <psoc_global_conf_regs.h>
-#include <psoc_gpio0_regs.h>
 
 
 #include <bitops.h>
 #include <preboot.h>
-#include <pll.h>
 #include <uart.h>
 
 #include "common.h"
@@ -23,52 +21,10 @@
 	return 1;
 }
 
-u64 hl_pll_timestamp_freq_get(void)
-{
-	/* As per Gaudi2 PLL cfg.xslx, CPU PLL Div[3] drives CPU timestamp */
-	return hl_pll_get_freq(mmPSOC_CPU_PLL_CTRL_BASE, HL_PLL_DIV3);
-}
-
-void hl_uart_init(void)
-{
-	/* As per Gaudi2 PLL cfg.xlsx, CPU PLL Div[3] drives uart */
-	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_CPU_PLL_CTRL_BASE,
-					    HL_PLL_DIV3);
-
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
-	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
-}
-
-/* TODO:configure yellow LED(GPIO2) in output mode and turn on(active low) */
-void preboot_led_on(void)
-{
-	/* TODO: Check this */
-	u32 val = BIT(2);
-
-	/* configure GPIO in output mode */
-	regmap_writel(val, mmPSOC_GPIO0_OUTENSET);
-
-	/* Turn ON LED (pin is active low) */
-	val = regmap_readl(mmPSOC_GPIO0_DATAOUT);
-	val &= ~BIT(2);
-	regmap_writel(val, mmPSOC_GPIO0_DATAOUT);
-}
-
 /* initialize sram scrambling before u-boot is downloaded */
 void sram_scrambling_init(void)
 {
 	/* TODO */
-}
-u32 hl_spi_baudr_get(void)
-{
-	/* As per Gaudi2 PLL cfg.xlsx, PCI PLL Div[3] drives SPI */
-	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_CPU_PLL_CTRL_BASE,
-					   HL_PLL_DIV3);
-
-	u32 factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
-
-	return factor > 2 ? factor : 2;
 }
 
 void hl_ram_init(void)
diff --git a/arch/gaudi2/arm/asic.c b/arch/gaudi2/arm/asic.c
index a9b70f7..776c8f4 100644
--- a/arch/gaudi2/arm/asic.c
+++ b/arch/gaudi2/arm/asic.c
@@ -5,15 +5,9 @@
  *
  */
 
-#include <gaudi2_blocks_linux_driver.h>
-#include <psoc_global_conf_regs.h>
-#include <psoc_gpio0_regs.h>
-
 #include <string.h>
 #include <bitops.h>
 #include <preboot.h>
-#include <pll.h>
-#include <uart.h>
 #include <sha256.h>
 #include <boot_desc.h>
 #include <pci.h>
@@ -24,49 +18,6 @@
 {
 	/* TODO: Use DumpSel[*] to skip PCIe initialization */
 	return 0;
-}
-
-u64 hl_pll_timestamp_freq_get(void)
-{
-	/* As per Gaudi2 PLL cfg.xslx, CPU PLL Div[3] drives CPU timestamp */
-	return hl_pll_get_freq(mmPSOC_CPU_PLL_CTRL_BASE, HL_PLL_DIV3);
-}
-
-void hl_uart_init(void)
-{
-	/* As per Gaudi2 PLL cfg.xlsx, CPU PLL Div[3] drives uart */
-	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_CPU_PLL_CTRL_BASE,
-					    HL_PLL_DIV3);
-
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
-	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
-	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
-}
-
-u32 hl_spi_baudr_get(void)
-{
-	u32 factor;
-	/* As per Gaudi2 PLL cfg.xlsx, PCI PLL Div[3] drives SPI */
-	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_CPU_PLL_CTRL_BASE,
-					   HL_PLL_DIV3);
-
-	factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
-
-	return factor > 2 ? factor : 2;
-}
-
-/* configure yellow LED(GPIO2) in output mode and turn on(active low) */
-void preboot_led_on(void)
-{
-	u32 val = BIT(2);
-
-	/* configure GPIO in output mode */
-	regmap_writel(val, mmPSOC_GPIO0_OUTENSET);
-
-	/* Turn ON LED (pin is active low) */
-	val = regmap_readl(mmPSOC_GPIO0_DATAOUT);
-	val &= ~BIT(2);
-	regmap_writel(val, mmPSOC_GPIO0_DATAOUT);
 }
 
 bool ppboot_is_valid_img_desc(struct boot_desc *desc)
diff --git a/arch/gaudi2/common/asic.c b/arch/gaudi2/common/asic.c
index f593276..357d391 100644
--- a/arch/gaudi2/common/asic.c
+++ b/arch/gaudi2/common/asic.c
@@ -5,4 +5,54 @@
  *
  */
 
-/* TODO: Gaudi2 asic(HL_TARGET) specific common functions to be moved here */
+#include <gaudi2_blocks_linux_driver.h>
+#include <psoc_global_conf_regs.h>
+#include <psoc_gpio0_regs.h>
+
+#include <preboot.h>
+#include <pll.h>
+#include <uart.h>
+
+u64 hl_pll_timestamp_freq_get(void)
+{
+	/* As per Gaudi2 PLL cfg.xslx, CPU PLL Div[3] drives CPU timestamp */
+	return hl_pll_get_freq(mmPSOC_CPU_PLL_CTRL_BASE, HL_PLL_DIV3);
+}
+
+void hl_uart_init(void)
+{
+	/* As per Gaudi2 PLL cfg.xlsx, CPU PLL Div[3] drives uart */
+	u32 uart_clk_freq = hl_pll_get_freq(mmPSOC_CPU_PLL_CTRL_BASE,
+					    HL_PLL_DIV3);
+
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_8);	/* UART0 RXD */
+	regmap_writel(1, mmPSOC_GLOBAL_CONF_PAD_SEL_9);	/* UART0 TXD */
+	preboot_uart_init(uart_clk_freq / HL_UART_BAUDRATE);
+}
+
+/* TODO:configure yellow LED(GPIO2) in output mode and turn on(active low) */
+void preboot_led_on(void)
+{
+	/* TODO: Check this */
+	u32 val = BIT(2);
+
+	/* configure GPIO in output mode */
+	regmap_writel(val, mmPSOC_GPIO0_OUTENSET);
+
+	/* Turn ON LED (pin is active low) */
+	val = regmap_readl(mmPSOC_GPIO0_DATAOUT);
+	val &= ~BIT(2);
+	regmap_writel(val, mmPSOC_GPIO0_DATAOUT);
+}
+
+u32 hl_spi_baudr_get(void)
+{
+	u32 factor;
+	/* As per Gaudi2 PLL cfg.xlsx, PCI PLL Div[3] drives SPI */
+	u32 spi_clk_freq = hl_pll_get_freq(mmPSOC_CPU_PLL_CTRL_BASE,
+					   HL_PLL_DIV3);
+
+	factor = spi_clk_freq / HL_SPI_FLASH_MAX_FREQ;
+
+	return factor > 2 ? factor : 2;
+}
